extra_data/reader.py

```diff
@@ -1633,6 +1633,107 @@ def __contains__(self, aliased_item):
         except KeyError:
             return False
 
+    def __repr__(self):
+        """
+        Pretty-print all the aliases.
+        """
+        RED = "\033[91m"
+        END_COLOR = "\033[0m"
+
+        # Check if the alias exists
+        def alias_exists(alias):
+            try:
+                self[alias]
+                return True
+            except KeyError:
+                return False
+
+        # Get the right icon for an alias
+        def alias_icon(exists):
+            if isinstance(exists, str):
+                exists = alias_exists(exists)
+
+            return " " if exists else f"{RED}✗{END_COLOR}"
+
+        # Find the alias for a source, if one exists
+        def source_alias(source):
```

Comment:
I wonder if such a method may actually be public, `run.alias.search('<source>')`
Ultimately this PR also offers a way to do this, but at some point large alias definition may becomes unwieldy that people want to filter interactively in a notebook?